### ISSUE TITLE
Issue with Migrations and Delta Callbacks

### ISSUE THREAD

**Issue by rywall** [pat/thinking-sphinx#1073]
I've run into an order of operations issue.

1. Create a Rails migration (it can be empty for replication purposes).
2. Create another Rails migration that creates a table for a model you want to index.
3. Create an ActiveRecord Sphinx index with delta indexing for that model.
4. run `rake db:migrate`

The rake task fails after the first migration because `ActiveRecord::SchemaMigration.create` is called to insert a record into the schema migrations table after the first migration is complete. 

This in turn triggers the `before_save` TS delta callbacks which eventually tries to load the SQL table underlying the index you just created. That table hasn't been created yet because it is in the 2nd migration.

Here is a backtrace:

```
Mysql2::Error: Table 'sphinx_test.pages' doesn't exist
gems/mysql2-0.4.9/lib/mysql2/client.rb:120:in `_query'
gems/mysql2-0.4.9/lib/mysql2/client.rb:120:in `block in query'
gems/mysql2-0.4.9/lib/mysql2/client.rb:119:in `handle_interrupt'
gems/mysql2-0.4.9/lib/mysql2/client.rb:119:in `query'
gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:317:in `block in execute'
gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract_adapter.rb:484:in `block in log'
gems/activesupport-4.2.10/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract_adapter.rb:478:in `log'
gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:317:in `execute'
gems/activerecord-4.2.10/lib/active_record/connection_adapters/mysql2_adapter.rb:217:in `execute'
gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:324:in `execute_and_free'
gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:471:in `columns'
gems/activerecord-4.2.10/lib/active_record/connection_adapters/schema_cache.rb:43:in `columns'
gems/activerecord-4.2.10/lib/active_record/attributes.rb:93:in `columns'
gems/activerecord-4.2.10/lib/active_record/model_schema.rb:260:in `column_names'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/sql_source/template.rb:43:in `inheriting?'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/sql_source/template.rb:33:in `class_column'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/sql_source/template.rb:9:in `apply'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/sql_source.rb:27:in `initialize'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/index.rb:8:in `new'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/index.rb:8:in `append_source'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/interpreter.rb:64:in `__source'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/interpreter.rb:20:in `indexes'
app/indices/page_index.rb:2:in `block in <top (required)>'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/core/interpreter.rb:3:in `translate!'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/core/index.rb:44:in `interpret_definition!'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/index.rb:36:in `sources'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/configuration/consistent_ids.rb:31:in `collect'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/configuration/consistent_ids.rb:31:in `sources'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/configuration/consistent_ids.rb:19:in `attributes'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/configuration/consistent_ids.rb:23:in `sphinx_internal_ids'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/configuration/consistent_ids.rb:7:in `reconcile'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/configuration.rb:174:in `normalise'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/configuration.rb:87:in `block in preload_indices'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/configuration.rb:78:in `synchronize'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/configuration.rb:78:in `preload_indices'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/index_set.rb:30:in `all_indices'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/index_set.rb:66:in `indices_for_references'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/index_set.rb:61:in `indices'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/index_set.rb:9:in `each'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/callbacks/delta_callbacks.rb:36:in `select'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/callbacks/delta_callbacks.rb:36:in `delta_indices'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/callbacks/delta_callbacks.rb:40:in `delta_indices?'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/active_record/callbacks/delta_callbacks.rb:19:in `before_save'
bundler/gems/thinking-sphinx-5784551c3b61/lib/thinking_sphinx/callbacks.rb:7:in `block (2 levels) in callbacks'
gems/activesupport-4.2.10/lib/active_support/callbacks.rb:455:in `public_send'
gems/activesupport-4.2.10/lib/active_support/callbacks.rb:455:in `block in make_lambda'
gems/activesupport-4.2.10/lib/active_support/callbacks.rb:164:in `block in halting'
gems/activesupport-4.2.10/lib/active_support/callbacks.rb:504:in `block in call'
gems/activesupport-4.2.10/lib/active_support/callbacks.rb:504:in `each'
gems/activesupport-4.2.10/lib/active_support/callbacks.rb:504:in `call'
gems/activesupport-4.2.10/lib/active_support/callbacks.rb:92:in `__run_callbacks__'
gems/activesupport-4.2.10/lib/active_support/callbacks.rb:778:in `_run_save_callbacks'
gems/activerecord-4.2.10/lib/active_record/callbacks.rb:302:in `create_or_update'
gems/activerecord-4.2.10/lib/active_record/persistence.rb:142:in `save!'
gems/activerecord-4.2.10/lib/active_record/validations.rb:43:in `save!'
gems/activerecord-4.2.10/lib/active_record/attribute_methods/dirty.rb:29:in `save!'
gems/activerecord-4.2.10/lib/active_record/transactions.rb:291:in `block in save!'
gems/activerecord-4.2.10/lib/active_record/transactions.rb:351:in `block in with_transaction_returning_status'
gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `block in transaction'
gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract/transaction.rb:184:in `within_new_transaction'
gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
gems/activerecord-4.2.10/lib/active_record/transactions.rb:220:in `transaction'
gems/transaction_retry-1.0.2/lib/transaction_retry/active_record/base.rb:23:in `transaction_with_retry'
gems/activerecord-4.2.10/lib/active_record/transactions.rb:348:in `with_transaction_returning_status'
gems/activerecord-4.2.10/lib/active_record/transactions.rb:291:in `save!'
gems/activerecord-4.2.10/lib/active_record/persistence.rb:51:in `create!'
gems/activerecord-4.2.10/lib/active_record/migration.rb:1054:in `record_version_state_after_migrating'
gems/activerecord-4.2.10/lib/active_record/migration.rb:1024:in `block in execute_migration_in_transaction'
gems/activerecord-4.2.10/lib/active_record/migration.rb:1071:in `ddl_transaction'
gems/activerecord-4.2.10/lib/active_record/migration.rb:1022:in `execute_migration_in_transaction'
gems/activerecord-4.2.10/lib/active_record/migration.rb:984:in `block in migrate'
gems/activerecord-4.2.10/lib/active_record/migration.rb:980:in `each'
gems/activerecord-4.2.10/lib/active_record/migration.rb:980:in `migrate'
gems/activerecord-4.2.10/lib/active_record/migration.rb:823:in `up'
gems/activerecord-4.2.10/lib/active_record/migration.rb:801:in `migrate'
gems/activerecord-4.2.10/lib/active_record/tasks/database_tasks.rb:139:in `migrate'
```

**Comment by pat** [comment]
Just pushed a fix for this :) d5612207a9f828c05c9e65c166c86e3cec6b4e74

**Comment by rywall** [comment]
Awesome! Thanks @pat.
